### PR TITLE
[Feature] Embedded cue support.

### PR DESCRIPTION
**Pull request by PSyton**
This patch add full support for read music metadata from embedded cuesheets.

### CODE REVIEW

**Comment by PSyton** [comment]
Anybody can review this?

**Comment by FernetMenta** [comment]
why did you ignore my comments?

**Comment by PSyton** [comment]
I didn't ignore. All fixed.

File: xbmc/music/infoscanner/MusicInfoScanner.cpp [hunk 1]
```diff
@@ -538,14 +539,18 @@ INFO_RET CMusicInfoScanner::ScanTags(const CFileItemList& items, CFileItemList&
...
-      m_handle->SetPercentage(m_currentItem/(float)m_itemCount*100);
+      m_handle->SetPercentage(m_currentItem / (float)m_itemCount * 100);
 
     if (!tag.Loaded())
     {
       CLog::Log(LOGDEBUG, "%s - No tag found for: %s", __FUNCTION__, pItem->GetPath().c_str());
       continue;
     }
-    scannedItems.Add(pItem);
+    pItem->LoadEmbeddedCue();
```
Comment (FernetMenta) [inline xbmc/music/infoscanner/MusicInfoScanner.cpp]: I asked you to check the tag first. the code is hard to read this way. why jumping into a method called "LoadEmbeddedCue" if there is no embedded cue?

**Comment by PSyton** [comment]
Ok. I've added tag check but I don't think that it made code simpler.

**Comment by FernetMenta** [comment]
thanks. IMO it is much clearer now.
could you please squash the last 2 commits

**Comment by PSyton** [comment]
done

**Comment by FernetMenta** [comment]
I still see a commit named "Refactor" which is meaningless from our repo perspective. Code which does not exist can't be refactored.

**Comment by PSyton** [comment]
I have additional commits introducing ReplayGain handling for cuesheets. Should I append this patches to this PR or create new?

**Comment by FernetMenta** [comment]
please create new pr for the others
you have renamed the commit but I wanted it to be squashed. this pr introduces ProcessCueDocument in first commit, then you rename this method in 3rd commit. we don't want to merge the history of this change.

**Comment by PSyton** [comment]
Ok. Now it's all squashed.

**Comment by FernetMenta** [comment]
@PSyton  thanks, can you rebase and resolve conflicts please.

@Montellese mind having a look at this, in particular at the db related code.

**Comment by PSyton** [comment]
Rebased.

**Comment by FernetMenta** [comment]
jenkins build this please

**Comment by FernetMenta** [comment]
@PSyton apart from win32 build failed for all platforms. click the "details" link and check the build outputs for the platforms. i.e. Linux shows: http://jenkins.kodi.tv/job/LINUX-64/3158/console

**Comment by PSyton** [comment]
Now should build.

**Comment by FernetMenta** [comment]
jenkins build this please

**Pull request merged**